listenbrainz/webserver/static/js/src/recommendations/Recommendations.tsx

```diff
@@ -0,0 +1,397 @@
+/* eslint-disable jsx-a11y/anchor-is-valid,camelcase */
+
+import { AlertList } from "react-bs-notifier";
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { isEqual, get } from "lodash";
+
+import BrainzPlayer from "../BrainzPlayer";
+import APIService from "../APIService";
+import Loader from "../components/Loader";
+import RecommendationCard from "./RecommendationCard";
+
+export interface RecommendationsProps {
+  apiUrl: string;
+  recommendations?: Array<Recommendation>;
+  profileUrl?: string;
+  spotify: SpotifyUser;
+  user: ListenBrainzUser;
+  webSocketsServerUrl: string;
+  currentUser?: ListenBrainzUser;
+}
+
+export interface RecommendationsState {
+  alerts: Array<Alert>;
+  currentRecommendation?: Recommendation;
+  direction: BrainzPlayDirection;
+  recommendations: Array<Recommendation>;
+  loading: boolean;
+  currRecPage?: number;
+  totalRecPages: number;
+  recommendationFeedbackMap: RecommendationFeedbackMap;
+}
+
+export default class Recommendations extends React.Component<
+  RecommendationsProps,
+  RecommendationsState
+> {
+  private brainzPlayer = React.createRef<BrainzPlayer>();
+  private recommendationsTable = React.createRef<HTMLTableElement>();
+  private APIService: APIService;
+
+  private expectedRecommendationsPerPage = 25;
+
+  constructor(props: RecommendationsProps) {
+    super(props);
+    this.state = {
+      alerts: [],
+      recommendations:
+        props.recommendations?.slice(0, this.expectedRecommendationsPerPage) ||
+        [],
+      loading: false,
+      direction: "down",
+      currRecPage: 1,
+      totalRecPages: props.recommendations
+        ? Math.ceil(
+            props.recommendations.length / this.expectedRecommendationsPerPage
+          )
+        : 0,
+      recommendationFeedbackMap: {},
+    };
+
+    this.recommendationsTable = React.createRef();
+    this.APIService = new APIService(
+      props.apiUrl || `${window.location.origin}/1`
+    );
+  }
+
+  componentDidMount(): void {
+    const { user, currentUser } = this.props;
+    const { currRecPage } = this.state;
+    if (currentUser?.name === user?.name) {
+      this.loadFeedback();
+    }
+    window.history.replaceState(null, "", `?page=${currRecPage}`);
+  }
+
+  getFeedback = async () => {
+    const { user, recommendations } = this.props;
```

Comment:
Follow-up to my previous comment, this should be :
```
const { user } = this.props;
const { recommendations } = this.state;
```
Since the recommendations in the state at any given time are the recs on the page the user is on.

Then we'll need to call `loadFeedback` on each page change as well.